module/zfs/spa.c

```diff
@@ -2316,6 +2320,186 @@ vdev_count_verify_zaps(vdev_t *vd)
 }
 #endif
 
+static unsigned long
+spa_get_hostid(void)
+{
+	unsigned long myhostid;
+
+#ifdef	_KERNEL
+	myhostid = zone_get_hostid(NULL);
+#else	/* _KERNEL */
+	/*
+	 * We're emulating the system's hostid in userland, so
+	 * we can't use zone_get_hostid().
+	 */
+	(void) ddi_strtoul(hw_serial, NULL, 10, &myhostid);
+#endif	/* _KERNEL */
+
+	return (myhostid);
+}
+
+/*
+ * Determine whether the activity check is required.
+ */
+static boolean_t
+spa_activity_check_required(spa_t *spa, uberblock_t *ub, nvlist_t *config)
+{
+	uint64_t config_pool_state = 0;
+	uint64_t hostid = 0, myhostid = spa_get_hostid();
+	uint64_t orig_txg = 0;
```

Comment:
Let's rename `orig_txg` and `orig_timestamp` to `tryconfig_txg` and `tryconfig_timestamp` which I find much less confusing.